pkg/operator/controller/controller.go

```diff
@@ -79,74 +76,66 @@ type reconciler struct {
 // namespace, and will do all the work to ensure the clusteringress is in the
 // desired state.
 func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	errs := []error{}
 	result := reconcile.Result{}
 
-	// TODO: Should this be another controller?
-	defer func() {
-		err := r.syncOperatorStatus()
-		if err != nil {
-			log.Error(err, "failed to sync operator status")
-		}
-	}()
-
 	log.Info("reconciling", "request", request)
 
 	// Get the current ingress state.
 	ingress := &ingressv1alpha1.ClusterIngress{}
-	err := r.Client.Get(context.TODO(), request.NamespacedName, ingress)
-	if err != nil {
+	if err := r.Client.Get(context.TODO(), request.NamespacedName, ingress); err != nil {
 		if errors.IsNotFound(err) {
 			// This means the ingress was already deleted/finalized and there are
 			// stale queue entries (or something edge triggering from a related
 			// resource that got deleted async).
 			log.Info("clusteringress not found; reconciliation will be skipped", "request", request)
 		} else {
-			// Try again later.
-			log.Error(err, "failed to get clusteringress", "request", request)
-			result.RequeueAfter = 10 * time.Second
+			errs = append(errs, fmt.Errorf("failed to get clusteringress %q: %v", request, err))
 		}
 		ingress = nil
 	}
 
 	caSecret, err := r.ensureRouterCACertificateSecret()
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure CA secret: %v", err)
+		errs = append(errs, fmt.Errorf("failed to ensure CA secret: %v", err))
+	} else {
+		// TODO: This should be in a different reconciler as it's independent of an
+		// individual ingress. We only really need to trigger this when a
+		// clusteringress is added or deleted...
+		// Find all clusteringresses to compute CA states.
+		ingresses := &ingressv1alpha1.ClusterIngressList{}
+		if err := r.Client.List(context.TODO(), &client.ListOptions{Namespace: r.Namespace}, ingresses); err != nil {
+			errs = append(errs, fmt.Errorf("failed to list clusteringresses in namespace %s: %v", r.Namespace, err))
+		} else if err := r.ensureRouterCAConfigMap(caSecret, ingresses.Items); err != nil {
```

Comment:
I don't understand why `ensureRouterCAConfigMap` was moved. Doesn't the outcome of the *current ingress reconcile* affect the outcome of `ensureRouterCAConfigMap`?

Reply:
It shouldn't.  We do need to mind the logic around `ensureRouterCAConfigMap` if we change the `.spec.defaultCertificateSecret` defaulting logic, but it shouldn't be an issue otherwise.

The only change in behavior from moving `ensureRouterCAConfigMap` is that we publish the `router-ca` configmap before we reconcile the clusteringress.  With the other changes, we publish the `router-ca` configmap even if the clusteringress reconciliation fails, so this only changes the order in which the objects are created.